src/ServiceStack/Validation/ValidationFilters.cs

```diff
@@ -62,9 +65,74 @@ public static async Task RequestFilterAsync(IRequest req, IResponse res, object
             }
             finally
             {
-                using (validator as IDisposable) { }
+                using (validator as IDisposable)
+                {
+                }
+            }
+        }
+
+        public static async Task ResponseFilterAsync(IRequest req, IResponse res, object requestDto)
+        {
+            var response = requestDto as IHasResponseStatus;
+            if (response == null)
+                return;
+
+            var validator = ValidatorCache.GetValidator(req, req.Dto.GetType());
+            if (validator == null)
+                return;
+
+            var validationResult = await Validate(validator, req, req.Dto);
+
+            var responseStatus = response.ResponseStatus ??
+                                 DtoUtils.CreateResponseStatus(validationResult.Errors[0].ErrorCode);
+            foreach (var error in validationResult.Errors)
+            {
+                var responseError = new ResponseError
+                {
+                    ErrorCode = error.ErrorCode,
+                    FieldName = error.PropertyName,
+                    Message = error.ErrorMessage,
+                    Meta = new Dictionary<string, string> {["Severity"] = error.Severity.ToString()}
+                };
+                responseStatus.Errors.Add(responseError);
+            }
+
+            response.ResponseStatus = responseStatus;
+        }
+
+        private static async Task<ValidationResult> Validate(IValidator validator, IRequest req, object requestDto)
+        {
+            var ruleSet = req.Verb;
+
+            try
```

Comment:
You don't need `try { ... } finally { using(validator as IDisposable){} }` You can just use `using (validator as IDisposable) { ... }`

Reply:
Removed the try finally block